src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/service/LimitedPartnerService.java

```diff
@@ -92,4 +97,16 @@ private Resource createLimitedPartnerTransactionResource(String submissionUri) {
 
         return limitedPartnerResource;
     }
+
+    public LimitedPartnerDto getLimitedPartner(Transaction transaction, String submissionId) throws ResourceNotFoundException {
+        String submissionUri = getSubmissionUri(transaction.getId(), submissionId);
+        if (!transactionUtils.isTransactionLinkedToLimitedPartnerSubmission(transaction, submissionUri)) {
+            throw new ResourceNotFoundException(String.format(
+                    "Transaction id: %s does not have a resource that matches submission id: %s", transaction, submissionId));
```

Comment:
You need to use `transaction.getId()` as the second parameter to `String.format()`.